base/int.jl

```diff
@@ -79,6 +79,13 @@ end
 
 checked_abs(x::Unsigned) = abs(x)
 
+function checked_neg{T<:Signed}(x::T)
+    x == typemin(T) && throw(OverflowError())
```

Comment:
this isn't true of all Signed types, only the twos-complement representation. for example, it wouldn't be true of IEEE float (if those were a julia subtype of Signed), ones-complement, or sign-magnitude.